src/components/ClientSideLoggingToolMenu/BaseClientSideLoggingToolMenu.tsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import {Alert} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import type {OnyxEntry} from 'react-native-onyx';
+import Button from '@components/Button';
+import Switch from '@components/Switch';
+import TestToolRow from '@components/TestToolRow';
+import Text from '@components/Text';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as Console from '@libs/actions/Console';
+import {parseStringifyMessages} from '@libs/Console';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {CapturedLogs, Log} from '@src/types/onyx';
+
+type BaseClientSideLoggingToolMenuOnyxProps = {
+    /** Logs captured on the current device */
+    capturedLogs: OnyxEntry<CapturedLogs>;
+
+    /** Whether or not logs should be stored */
+    shouldStoreLogs: OnyxEntry<boolean>;
+};
+
+type BaseClientSideLoggingToolProps = {
+    /** Locally created file */
+    file?: {path: string; newFileName: string; size: number};
+    /** Action to run when pressing Share button */
+    onShareLogs?: () => void;
+    /** Action to run when disabling the switch */
+    onDisableLogging: (logs: Log[]) => void;
+    /** Action to run when enabling logging */
+    onEnableLogging?: () => void;
+} & BaseClientSideLoggingToolMenuOnyxProps;
+
+function BaseClientSideLoggingToolMenu({shouldStoreLogs, capturedLogs, file, onShareLogs, onDisableLogging, onEnableLogging}: BaseClientSideLoggingToolProps) {
+    const onToggle = () => {
+        if (!shouldStoreLogs) {
+            Console.setShouldStoreLogs(true);
+
+            if (onEnableLogging) {
+                onEnableLogging();
+            }
+
+            return;
+        }
+
+        if (!capturedLogs) {
+            Alert.alert('No logs to share', 'There are no logs to share');
+            Console.disableLoggingAndFlushLogs();
+            return;
+        }
+
+        const logs = Object.values(capturedLogs);
+        const logsWithParsedMessages = parseStringifyMessages(logs);
+
+        onDisableLogging(logsWithParsedMessages);
+        Console.disableLoggingAndFlushLogs();
+    };
+    const styles = useThemeStyles();
+    return (
+        <>
+            <TestToolRow title="Client side logging">
```

Comment:
Should we add a translation for this phrase? I see that it's consistent with other titles in Tools (all are hardcoded English) so NAB, but curious if there's a reason for not translating